packages/core/src/__tests__/types.test.tsx

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable jest/expect-expect */
```

Comment:
Can you move the type-related tests to the `types.check.tsx` file? They don't need to be run with Jest since there are no runtime checks here and Jest won't report type errors that we're testing.